packages/shared_preferences/shared_preferences_platform_interface/test/shared_preferences_platform_interface_test.dart

```diff
@@ -3,19 +3,33 @@
 // found in the LICENSE file.
 
 import 'package:flutter_test/flutter_test.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
 
 void main() {
   TestWidgetsFlutterBinding.ensureInitialized();
 
   group(SharedPreferencesStorePlatform, () {
     test('disallows implementing interface', () {
-      expect(
-        () {
-          SharedPreferencesStorePlatform.instance = IllegalImplementation();
-        },
-        throwsAssertionError,
-      );
+      expect(() {
+        SharedPreferencesStorePlatform.instance = IllegalImplementation();
```

Comment:
I can't comment below, but IllegalImplementation should have a docstring explaining why it's illegal.

Reply:
Done.